### PR TITLE
Fix code gen generator

### PR DESCRIPTION
**Pull request by emlyn-rice**
Add add target executor

### CODE REVIEW

File: libs/nx-plugin/generators.json [hunk 1]
```diff
@@ -3,7 +3,12 @@
         "code-gen": {
             "factory": "./src/generators/code-gen/generator",
             "schema": "./src/generators/code-gen/schema.json",
-            "description": "code-gen generator"
+            "description": "Generates FeatureBoard types for your chosen language"
+        },
+        "AddCodeGenTarget": {
```
Comment (JakeGinnivan) [inline libs/nx-plugin/generators.json]: Should this be kebab case?

File: libs/nx-plugin/project.json [hunk 2]
```diff
@@ -31,7 +31,8 @@
             "options": {
                 "passWithNoTests": false,
                 "reportsDirectory": "../../coverage/libs/nx-plugin"
-            }
+            },
+            "dependsOn": ["^build"]
```
Comment (JakeGinnivan) [inline libs/nx-plugin/project.json]: Why does test require build?

File: libs/nx-plugin/src/executors/code-gen/executor.ts [hunk 3]
```diff
@@ -4,19 +4,30 @@ import {
...
-    { projectName, ...options }: CodeGenExecutorSchema,
+    options: CodeGenExecutorSchema,
+    context: ExecutorContext,
+    tree?: Tree,
 ) {
-    const project = readProjectConfiguration(tree, projectName)
-    const dryRun = isDryRun()
-    const verbose = isVerbose()
+    if (!context.projectName) {
+        throw new Error('No projectName')
```
Comment (MatthewJNixon) [inline libs/nx-plugin/src/executors/code-gen/executor.ts]: nit: this could be more wordy and friendly perhaps?

**Review by MatthewJNixon — APPROVED** [review]

File: libs/nx-plugin/generators.json (hunk 1 above)
Comment (emlyn-rice) [reply libs/nx-plugin/generators.json]: Fixed

File: libs/nx-plugin/project.json (hunk 2 above)
Comment (emlyn-rice) [reply libs/nx-plugin/project.json]: Removed

File: libs/nx-plugin/src/executors/code-gen/executor.ts (hunk 3 above)
Comment (emlyn-rice) [reply libs/nx-plugin/src/executors/code-gen/executor.ts]: Updated

**Pull request merged**